src/components/ModalWizard/ModalWizard.js

```diff
@@ -0,0 +1,192 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+  noop,
+  bindMethods,
+  controlled,
+  Modal,
+  Wizard,
+  Icon,
+  Button
+} from '../../index';
+import ModalWizardBody from './ModalWizardBody';
+
+class BaseModalWizard extends React.Component {
+  constructor() {
+    super();
+    bindMethods(this, ['onBackClick', 'onNextClick', 'goToStep']);
+  }
+
+  onBackClick() {
+    this.goToStep(Math.max(this.props.activeStepIndex - 1, 0));
+  }
+
+  onNextClick() {
+    const { steps, activeStepIndex } = this.props;
+    this.goToStep(Math.min(activeStepIndex + 1, steps.length - 1));
+  }
+
+  goToStep(newStepIndex) {
+    const {
+      setControlledState,
+      onStepChanged,
+      activeStepIndex,
+      onNext,
+      onBack,
+      steps
+    } = this.props;
+    const currentStep = steps[activeStepIndex];
+    setControlledState({ activeStepIndex: newStepIndex });
+    if (onStepChanged) onStepChanged(newStepIndex);
```

Comment:
onStepChanged should validate with the application prior to actually navigating. There are implementations where the next button (or navigating forward) causes some action which may need to be validated before navigation. The onStepChanged method should be able to return a boolean or a promise to validate the navigation.

Reply:
Agreed, @jeff-phillips-18. we had implemented that validation with other props, like a `disableGoto` or a `shouldPreventNavigation` callback or boolean in each step of the steps array. or a callback at the component level that takes a step index argument and returns that boolean.  onStepChanged returning a promise is interesting-- but right now it's intended to be a callback for after the step changes, and won't be called at all if the step is prevented from changing. I'm cool with either pattern.

Reply:
To be clear, the `disableGoto` was added to v2v after this refactor forked off-- i only didn't add it here because we ended up disabling it anyway with `stepButtonsDisabled`. All the validation for navigating forward is currently on `shouldDisableNextStep`, a callback that you can pass in to `ModalWizard` that takes a step index. I was considering also making available a `nextStepDisabled` prop that could override that. I think that might be sufficient instead of using promises on the optional callbacks.

Reply:
In case the user wants to add a completely custom button to the wizard's footer and/or further tweak existing button logic and/or add other custom stuff into the footer, it should also be possible to replace the footer with a custom implementation (in such case, the responsibility of handling footer content is shifted from `ModalWizard` to its user).

Reply:
Good point @vojtechszocs. I'll have to add that into my next iteration.